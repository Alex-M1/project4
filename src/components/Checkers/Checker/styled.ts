import { IStyled } from 'common/types/styledTypes';
import { changeTheme } from 'src/helpers/changeTheme';
import styled from 'styled-components';

interface IQueen extends IStyled {
  isQueen: boolean;
}

export const StCellWhite = styled.div <IQueen>`
  border: 2px solid ${(props) => changeTheme(props, 'black')};
  box-shadow: 0 0 2px 2px ${(props) => changeTheme(props, 'black')};
  background: ${(props) => changeTheme(props, 'white')};
  &::after {
    border: 2px solid ${(props) => changeTheme(props, 'black')};
    background: ${(props) => (props.isQueen ? changeTheme(props, 'gold') : changeTheme(props, 'white'))};
  }
`;
export const StCellBlack = styled.div <IQueen>`
  border: 2px solid ${(props) => changeTheme(props, 'black')};
  box-shadow: 0 0 2px 2px ${(props) => changeTheme(props, 'black')};
  background: ${(props) => (props.isQueen ? changeTheme(props, 'gold') : changeTheme(props, 'blackCellBg'))};
  &::after {
    border: 2px solid ${(props) => changeTheme(props, 'black')};
    background: ${(props) => changeTheme(props, 'blackCellBg')};
  }
`;
