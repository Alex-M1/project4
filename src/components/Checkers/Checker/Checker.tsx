// import { CHECKER_FIELD_INIT } from 'constants/constants';
import React from 'react';
import { useDrag } from 'react-dnd';
import { useTheme } from 'src/components/hooks/useTheme';
import { StCellBlack, StCellWhite, StOpacity } from './styled';

interface IProps {
  isBlack?: boolean,
}

const Checker: React.FC<IProps> = ({ isBlack }) => {
  const { colors, theme } = useTheme();
  const [{ isDragging }, drag] = useDrag(() => ({
    type: 'checker',
    collect: (monitor) => ({
      isDragging: !!monitor.isDragging(),
    }),
  }));
  return (
    <StOpacity ref={drag} isDragging={isDragging}>
      {
        // if (isBlack) {
        //   if (CHECKER_FIELD_INIT)
        // }
        isBlack
          ? <StCellBlack colors={colors} theme={theme} />
          : <StCellWhite colors={colors} theme={theme} />
      }
    </StOpacity>
  );
};
export default Checker;
