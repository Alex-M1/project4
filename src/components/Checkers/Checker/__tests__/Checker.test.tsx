import React from 'react';
import { shallow, mount } from 'enzyme';
import Checker from '../Checker';

describe('Checker', () => {
  describe('white checker', () => {
    let props;
    beforeEach(() => {
      props = {
        isBlack: false,
        isQueen: false,
      };
    });
    it('Should match snapshot', () => {
      const component = shallow(<Checker {...props} />);
      expect(component.html()).toMatchSnapshot();
    });
    it('should render div', () => {
      const component = mount(<Checker {...props} />);
      expect(component.find('div')).toHaveLength(2);
    });
    it('should render StCellWhite', () => {
      const component = mount(<Checker {...props} />);
      expect(component.find('styled__StCellWhite')).toHaveLength(1);
    });
    it('should render StCellWhite & queen', () => {
      const newProps = {
        isBlack: false,
        isQueen: true,
      };
      const component = mount(<Checker {...newProps} />);
      expect(component.find('styled__StCellWhite')).toHaveLength(1);
    });
  });
  describe('black checker', () => {
    let props;
    beforeEach(() => {
      props = {
        isBlack: true,
        isQueen: false,
      };
    });
    it('Should match snapshot', () => {
      const component = shallow(<Checker {...props} />);
      expect(component.html()).toMatchSnapshot();
    });
    it('should render div', () => {
      const component = mount(<Checker {...props} />);
      expect(component.find('div')).toHaveLength(2);
    });
    it('should render StCellBlack', () => {
      const component = mount(<Checker {...props} />);
      expect(component.find('styled__StCellBlack')).toHaveLength(1);
    });
    it('should render StCellWhite & queen', () => {
      const newProps = {
        isBlack: true,
        isQueen: true,
      };
      const component = mount(<Checker {...newProps} />);
      expect(component.find('styled__StCellBlack')).toHaveLength(1);
    });
  });
});
