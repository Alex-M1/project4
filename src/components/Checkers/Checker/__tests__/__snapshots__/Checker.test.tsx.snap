// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Checker black checker Should match snapshot 1`] = `"<div><div class=\\"styled__StCellBlack-gTsvAF fsoxms\\"></div></div>"`;

exports[`Checker white checker Should match snapshot 1`] = `"<div><div class=\\"styled__StCellWhite-eopeRB czkzjO\\"></div></div>"`;
